### PR TITLE
add protocol and executor version to replay

### PR DESCRIPTION
**Pull request by laura-makdah**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui/src/client_commands.rs [hunk 1]
```diff
@@ -217,19 +215,6 @@ pub enum SuiClientCommands {
...
-    Faucet {
-        /// Address (or its alias)
-        #[clap(long)]
-        #[arg(value_parser)]
-        address: Option<KeyIdentity>,
-        /// The url to the faucet
-        #[clap(long)]
-        url: Option<String>,
-    },
-
```
Comment (stefan-mysten) [inline crates/sui/src/client_commands.rs:232]: This is the faucet command that we've been advertising, so not sure what went wrong here, but this needs to stay.

File: crates/sui/src/client_commands.rs [hunk 2]
```diff
@@ -741,12 +728,9 @@ impl SuiClientCommands {
...
                 profile_output,
             } => {
-                move_vm_profiler::gas_profiler_feature_disabled! {
-                    bail!(
-                        "gas-profiler feature is not enabled, rebuild or reinstall with \
-                         --features gas-profiler"
-                    );
-                };
+                if !cfg!(feature = "gas-profiler") {
+                    bail!("gas-profiler feature is not enabled, rebuild or reinstall with --features gas-profiler");
```
Comment (stefan-mysten) [inline crates/sui/src/client_commands.rs:732]: Let's give proper error messages.
```suggestion
                    bail!("The gas-profiler feature is not enabled for this CLI build. Please rebuild or reinstall with --features gas-profiler.");
```

File: crates/sui/src/client_commands.rs [hunk 3]
```diff
@@ -1406,7 +1369,7 @@ impl SuiClientCommands {
                     ));
                 }
 
-                if let Some(address) = address {
+                if let Some(address) = address.clone() {
```
Comment (stefan-mysten) [inline crates/sui/src/client_commands.rs:1372]: Why is this clone needed?

File: crates/sui/src/client_commands.rs [hunk 4]
```diff
@@ -1909,7 +1872,6 @@ impl Display for SuiClientCommandResult {
             SuiClientCommandResult::ReplayTransaction => {}
             SuiClientCommandResult::ReplayBatch => {}
             SuiClientCommandResult::ReplayCheckpoints => {}
-            SuiClientCommandResult::NoOutput => {}
```
Comment (stefan-mysten) [inline crates/sui/src/client_commands.rs:1912]: Hmm, something went wrong, maybe a rebase will fix things?

File: crates/sui/src/client_commands.rs [hunk 5]
```diff
@@ -2069,6 +2031,13 @@ pub struct ObjectOutput {
 
 impl From<&SuiObjectData> for ObjectOutput {
     fn from(obj: &SuiObjectData) -> Self {
+        let owner_type = match obj.owner {
+            Some(Owner::AddressOwner(_)) => Some("AddressOwner".to_string()),
+            Some(Owner::ObjectOwner(_)) => Some("ObjectOwner".to_string()),
+            Some(Owner::Shared { .. }) => Some("Shared".to_string()),
+            Some(Owner::Immutable) => Some("Immutable".to_string()),
+            None => None,
+        };
```
Comment (stefan-mysten) [inline crates/sui/src/client_commands.rs:2040]: This was fixed by someone else in a PR that I merged. Please rebase, this should be different.

**Review by stefan-mysten — CHANGES_REQUESTED** [review]

File: crates/sui/src/client_commands.rs (hunk 4 above)
Comment (laura-makdah) [reply crates/sui/src/client_commands.rs:1912]: this is super weird some changes I did not make got added to this pr

**Pull request closed without merging**